File: testdata/precision-suite-clojure/clean_datablob.clj
(ns app.datablob
  "Clean: data blobs and identifiers whose long alphanumeric runs look like
   secrets but are inert data — an embedded base64 data-URI, a git SHA, UUIDs, a
   hex color. lexctx classifies the string bodies as data, so the entropy/secret
   rules must not fire. Any finding here is a false positive.")

;; A base64 data-URI SVG icon — a long blob, not a credential.
(def icon
  "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgaGVpZ2h0PSIxNiI+PHBhdGggZD0iTTggMEwxNiA4TDggMTZMMCA4WiIvPjwvc3ZnPg==")

;; Public identifiers, not secrets.
(def build-sha "9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08")
(def request-id "550e8400-e29b-41d4-a716-446655440000")
(def brand-color "#1a2b3c")
(def sri-hash "sha384-oqVuAfXRKap7fdgcCY5uykM6+R9GqQ8K/uxy9rx7HNQlGYl1kPzQho1wx4JwY8wC")

File: testdata/precision-suite-php/clean_datablob.php
<?php
// Data-blob stressor: a long base64 data-URI embedded as a string literal. The
// long alphanumeric runs trip naive entropy/secret rules, but lexctx classifies
// the whole literal as a data blob, so nothing should fire. Zero findings.

$logo = 'data:image/svg+xml;base64,'
    . 'PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxMjgi'
    . 'IGhlaWdodD0iMTI4Ij48cmVjdCB3aWR0aD0iMTI4IiBoZWlnaHQ9IjEyOCIgZmlsbD0i'
    . 'IzAwN2FjYyIvPjx0ZXh0IHg9IjY0IiB5PSI3MCIgZm9udC1zaXplPSI0OCIgZmlsbD0i'
    . 'I2ZmZmIiB0ZXh0LWFuY2hvcj0ibWlkZGxlIj5OPC90ZXh0Pjwvc3ZnPg==';

// A hex color palette and a UUID — identifiers, not secrets.
$brand = '#007acc';
$requestId = '550e8400-e29b-41d4-a716-446655440000';

echo strlen($logo);
